src/helpers/useAxios.js

```diff
@@ -0,0 +1,26 @@
+import { useState, useEffect } from 'react';
+import axios from 'axios';
+
+const useAxios = (url, data) => {
+    const [response, setResponse] = useState(null);
+    const [error, setError] = useState(null);
+
+    const makeRequest = async () => {
+        try {
+            const res = await axios.post(url, data);
+            setResponse(res);
+        } catch (error) {
+            setError(error);
+        }
+    };
```

Comment:
Deze functie zal élke rerender opnieuw gemaakt worden (in memory), waardoor de useEffect ook opnieuw af zal gaan. Je kunt daarom twee dingen doen:

1. Je zet deze functie bínnen de useEffect functie
2. (Niet behandeld tijdens de les geloof ik): Je maakt gebruik van de [useCallback hook](https://reactjs.org/docs/hooks-reference.html#usecallback).